Extension/src/LanguageServer/client.ts

```diff
@@ -843,6 +846,20 @@ export class DefaultClient implements Client {
     private configStateReceived: ConfigStateReceived = { compilers: false, compileCommands: false, configProviders: undefined, timeout: false };
     private showConfigureIntelliSenseButton: boolean = false;
 
+    /** A queue of asynchronous tasks that need to be processed befofe ready is considered active.  */
+    private tasks = new Array<[ManualPromise<unknown>|undefined,() => Promise<unknown>]>();
```

Comment:
It might be nice to define an interface for this instead of using an array with unnamed fields. `[ManualPromise<unknown>|undefined,() => Promise<unknown>]`

Reply:
That's a [Tuple](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types), it is literally the typescript prescribed way to do this.

Reply:
I would disagree that TypeScript prescribes using tuples in this scenario (I didn't see it in your link).  But this is a personal preference suggestion.  I've had to read through enough unfamiliar code that used tuples to form an opinion against their use almost universally.  The lack of self-documenting fields is the biggest detractor for me as I review a lot of code and don't always have IntelliSense features available when I do that.  You could just as easily use an unnamed interface here and provide names for the fields so that you would know exactly what this data structure holds without having to search how it's constructed or used.

`{taskResult: ManualPromise<unknown> | undefined; task: () => Promise<unknown>}`